### PR TITLE
Enable Windows 11 OS version detection.

### LINKED ISSUE #12100: [Code] NppParameters::getWindowsVersion() can not detect Windows 11
### Description of the Issue

In [PowerEditor/src/Parameters.cpp](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/6d36d9d398497a6914c64a5804f0400b8bcd6fbd/PowerEditor/src/Parameters.cpp), the function `NppParameters::getWindowsVersion()`, which is meant to return the OS version the program is running on, can not return Windows 11 as the OS version detected. Instead, it will return `WV_UNKNOWN`.

### Expected Behavior

When the program is running on Windows 11, function `NppParameters::getWindowsVersion()` should return a separate symbol of the OS (e.g. add new item `WV_WIN11` to the `winVer` enumeration and return it).

### Actual Behavior

It will return WV_UNKNOWN.

### Why this is important
Returning WV_UNKNOWN on Windows 11 will breach the code logic. And since Windows 11 is already on the supported OS list, it will run on said OS.

### PR DESCRIPTION
**Pull request by LeonC510**
Fix #12100 by enabling Windows 11 version detection.

### CODE REVIEW

**Comment by donho** [comment]
I notice this PR is not from a new branch:
> Create a new branch for each PR. Make sure your branch name wasn't used before - you can add date (for example patch3_20200528) to ensure its uniqueness.

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

It's OK for the first time because there will no impact to the workflow of merging this PR. Please follow the suggestion for your next PR.

**Comment by donho** [comment]
It doesn't work under Windows 11:

image

File: PowerEditor/src/Parameters.cpp [hunk 1]
```diff
@@ -839,6 +839,9 @@ winVer NppParameters::getWindowsVersion()
    {
 		case VER_PLATFORM_WIN32_NT:
 		{
+			if (osvi.dwMajorVersion == 10 && osvi.dwMinorVersion == 22000)
```
Comment (ozone10) [inline PowerEditor/src/Parameters.cpp:842]: @donho @LeonC510 
this line is wrong
it should be `if (osvi.dwMajorVersion == 10 && osvi.dwMinorVersion == 0 && osvi.dwBuildNumber >= 22000)`

Comment (donho) [reply PowerEditor/src/Parameters.cpp:842]: @ozone10 
It works. Thank you for the tip!

**Comment by LeonC510** [comment]
> I notice this PR is not from a new branch:
> 
> > Create a new branch for each PR. Make sure your branch name wasn't used before - you can add date (for example patch3_20200528) to ensure its uniqueness.
> 
> https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests
> 
> It's OK for the first time because there will no impact to the workflow of merging this PR. Please follow the suggestion for your next PR.

Thank you for your warning. I will create a new branch for my next pull request.

**Comment by LeonC510** [comment]
> It doesn't work under Windows 11:
> 
> image

My mistake. @donho Really sorry for the inconvenience I caused.
And thank @ozone10  for pointing out the right way to do it!

**Comment by alankilborn** [comment]
@LeonC510 

>My mistake. Really sorry for the inconvenience I caused.

I think you can be forgiven.  There's a lot of conflicting information on how it is done when you look it up on the internet. :-)

**Pull request closed without merging**